src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -661,6 +661,29 @@ public SyntaxNode GetContainingTypeDeclaration(SyntaxNode root, int position)
                 .FirstOrDefault(n => n is BaseTypeDeclarationSyntax || n is DelegateDeclarationSyntax);
         }
 
+        public string GetNameOfContainingType(SyntaxNode node)
+        {
+            var containingType = node.Ancestors().FirstOrDefault(n => n is BaseTypeDeclarationSyntax || n is DelegateDeclarationSyntax);
+            if (containingType is BaseTypeDeclarationSyntax)
+            {
+                return ((BaseTypeDeclarationSyntax)containingType).GetNameToken().ValueText;
+            }
+            else if (containingType is DelegateDeclarationSyntax)
+            {
+                return ((DelegateDeclarationSyntax)containingType).GetNameToken().ValueText;
+            }
+            else
+            {
+                return "";
+            }
+        }
+
+        public string GetNameOfContainingMember(SyntaxNode node)
+        {
+            var containingMember = node.Ancestors().FirstOrDefault(n => n is MemberDeclarationSyntax);
```

Comment:
```suggestion
            var containingMember = node.Ancestors().OfType<MemberDeclarationSyntax>().FirstOrDefault();
```

Helps avoid cast below, and I'm _pretty sure_ it's just as fast